### PR TITLE
[FEATURE] Added option to edit regression tests

### PR DESCRIPTION
**Pull request by T1duS**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [X] I am an active contributor to the project.

---

### CODE REVIEW

File: tests/test_regression/TestControllers.py [hunk 1]
```diff
@@ -247,6 +247,16 @@ def test_edit_test(self):
...
         self.create_user_with_role(
             self.user.name, self.user.email, self.user.password, Role.admin)
+
+        for _ in range(3):
+            test = RegressionTest(1, '-autoprogram -out=ttxt -latin1 -2',
+                           InputType.file, OutputType.file, 1, 10)
+            test2 = RegressionTest(1, '-autoprogram -out=ttxt -latin1 -2',
+                           InputType.file, OutputType.file, 2, 10)
+            g.db.add(test)
+            g.db.add(test2)
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: You need to add these regression tests to categories. like category.regression_tests.append

Comment (T1duS) [reply tests/test_regression/TestControllers.py]: added

File: tests/test_regression/TestControllers.py [hunk 2]
```diff
@@ -232,11 +232,172 @@ def test_category_delete(self):
...
+            test = RegressionTest(1, '-autoprogram -out=ttxt -latin1 -2',
+                           InputType.file, OutputType.file, 1, 10)
+            test2 = RegressionTest(1, '-autoprogram -out=ttxt -latin1 -2',
+                           InputType.file, OutputType.file, 2, 10)
+            category = Category.query.filter(Category.id == 1).first()
+            category2 = Category.query.filter(Category.id == 2).first()
+            g.db.add(test)
+            g.db.add(test2)
+            category.regression_tests.append(test)
+            category2.regression_tests.append(test)
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: it will be test2

File: mod_regression/controllers.py [hunk 3]
```diff
@@ -83,10 +83,45 @@ def test_delete(regression_id):
...
+        test.input_type = InputType.from_string(form.input_type.data)
+        test.output_type = OutputType.from_string(form.output_type.data)
+
+        # adding test to its new category
+        category = Category.query.filter(Category.id == form.category_id.data).first()
+        category.regression_tests.append(test)
+
+        g.db.commit()
+        flash('Regression Test Updated')
+        return redirect(url_for('.index'))
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: I'd love to see the user being redirected to the test he edited before :+1:

**Pull request merged**